### PR TITLE
Update model docs to reStructuredText format

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR stems from the user doc extractor from @jougs and @terhorstd (Note #1355 probably should be merged first) 

It removes the doxygen style format and replaces it with reStructuredText in the models/ directory.
I also removed the SLI examples.

**Note** that now we've lost the automated linking (like in the See Also section) that breathe used to do.

see the current output here: https://nest-test.readthedocs.io/en/update_models_rst_format/from_cpp/index.html

### CODE REVIEW

**Comment by jougs** [comment]
@jessica-mitchell: I had a meeting with @sarakonradi yesterday, discussing what I would like to have changed in this one and we agreed that she will make the changes and send you a corresponding PR instead of me doing a review at this point. Once that happened, I'll have another closer look.

**Comment by jougs** [comment]
Damnit. While submitting the review, something went wrong and not even half of my comments made it here. I'll have a look again. Please don't start working on this. Sorry for the mess.

File: models/hh_cond_beta_gap_traub.h [hunk 1]
```diff
@@ -59,45 +59,42 @@ namespace nest
...
+hh_cond_beta_gap_traub - modified Hodgkin-Huxley neuron with added gap junction support and beta function synaptic conductance
+##############################################################################################################################
 
-Description:
+Description
++++++++++++
 
 hh_cond_beta_gap_traub is an implementation of a modified Hodgkin-Huxley model
-that also supports gap junctions.
+that also supports gap junction.
```
Comment (jougs) [inline models/hh_cond_beta_gap_traub.h]:
```suggestion
that also supports gap junctions.
```

File: models/iaf_psc_exp_htum.h [hunk 2]
```diff
@@ -131,17 +132,22 @@ the NEST source code (docs/model_details).
...
+
+SpikeEvent, CurrentEvent, DataLoggingRequest
 
 FirstVersion: March 2006
 
-Author: Moritz Helias
-*/
-class iaf_psc_exp_htum : public Archiving_Node
+EndUserDocs */
+class iaf_tum_2000 : public Archiving_Node
```
Comment (jougs) [inline models/iaf_psc_exp_htum.h]: Something's fishy here. Why is there a model name change in this PR? Is this a mixup with changes from another branch? I think the correct name should be `iaf_psc_exp_htum`.

Anyway, the (incomplete) change of name leads to the failure of the Travis tests. This should be sorted out.

File: models/amat2_psc_exp.h [hunk 3]
```diff
@@ -240,7 +244,7 @@ class amat2_psc_exp : public Archiving_Node
     /** Resting threshold in mV
         (relative to resting potential).
         The real resting threshold is (E_L_+omega_).
-        Called omega in [3]. */
+        Called omega in [3]_. */
```
Comment (jougs) [inline models/amat2_psc_exp.h]: I saw this change multiple times in this PR. And although it is consistent with the other changes, I'm not sure that this change is useful. The reason for my doubt is that this Doxygen style comment (`/** ... */`) will end up in the *developer* documentation, while the reference itself is defined in the *user* documentation.

The proper solution is most likely to
1. add/extend Doxygen class level documentation to point to the location of the user documentation for the model on readthedocs and mention that references are contained therein
1. make the references in the Doxygen comments throughout the model code be the same format as that of the rendered user documentation on readthedocs.

File: models/hh_psc_alpha_clopath.h [hunk 4]
```diff
@@ -57,16 +57,13 @@ namespace nest
...
-/** @BeginDocumentation
-@ingroup Neurons
-@ingroup hh
-@ingroup psc
-@ingroup clopath_n
+/* BeginUserDocs: neurons, Hodgkin-Huxley, post-synaptic current, Clopath synapse
 
-Name: hh_psc_alpha_clopath - Hodgkin-Huxley neuron model with support for the
-Clopath synapse.
+hh_psc_alpha_clopath - Hodgkin-Huxley neuron model with support for the Clopath synapse
```
Comment (jougs) [inline models/hh_psc_alpha_clopath.h]:
```suggestion
hh_psc_alpha_clopath - Hodgkin-Huxley neuron model with support for Clopath plasticity
```

File: models/hh_psc_alpha_clopath.h [hunk 5]
```diff
@@ -57,16 +57,13 @@ namespace nest
...
+#######################################################################################
```
Comment (jougs) [inline models/hh_psc_alpha_clopath.h]:
```suggestion
######################################################################################
```

File: models/ac_generator.h [hunk 6]
```diff
@@ -33,58 +33,70 @@
 #include "stimulating_device.h"
 #include "universal_data_logger.h"
 
+/* BeginUserDocs: devices, generator
 
-namespace nest
-{
+Produce an AC input current
```
Comment (jougs) [inline models/ac_generator.h]:
```suggestion
ac_generator - Produce an alternating current (AC) input
```

File: models/ac_generator.h [hunk 7]
```diff
@@ -33,58 +33,70 @@
...
+###########################
```
Comment (jougs) [inline models/ac_generator.h]:
```suggestion
########################################################
```

File: models/dc_generator.h [hunk 8]
```diff
@@ -38,35 +38,25 @@
 
 namespace nest
 {
-/** @BeginDocumentation
-@ingroup Devices
-@ingroup generator
+/* BeginUserDocs: devices, generator
 
-Name: dc_generator - provides DC input current
+dc_generator - provides DC input current
```
Comment (jougs) [inline models/dc_generator.h]:
```suggestion
dc_generator - Produce a direct current (DC) input
```

File: models/dc_generator.h [hunk 9]
```diff
@@ -38,35 +38,25 @@
...
+########################################
```
Comment (jougs) [inline models/dc_generator.h]:
```suggestion
##################################################
```

**Comment by jougs** [comment]
Can you please pull upstream/master again? Since the merge of #1245, there's a new MUSIC proxy whose documentation also needs conversion to RST. Thanks!

**Comment by heplesser** [comment]
@jessica-mitchell I just sent you a PR towards your branch to resolve the merge conflict.

**Comment by heplesser** [comment]
@jessica-mitchell This did not work quite as well as I had hope. Note that the iaf_chxk_2008 model has moved to the precise directory per #1396. Could you pull the most recent master into your branch? We can then sort out the model documentation to make sure that your changes to formatting and my changes to content get nicely combined.

**Comment by sarakonradi** [comment]
@heplesser The MUSIC proxy documentation also needs to be converted to RST. I assume that, once the ``iaf_chxk_2008`` merge conflict is solved, we will send a separate PR to @jessica-mitchell?

**Comment by jessica-mitchell** [comment]
@heplesser @sarakonradi  Merged master, seems like conflicts are gone :)

**Comment by sarakonradi** [comment]
#1484 updates the rate-based models' documentation and should be taken into account.

**Comment by heplesser** [comment]
> #1484 updates the rate-based models' documentation and should be taken into account.

@jessica-mitchell #1484 is merged into master now, so you could pull it directly in here. This may explain the new merge conflicts.

**Comment by sarakonradi** [comment]
Yes @heplesser, that definitely explains the new merge conflicts.

We should make sure the rate-based models and MUSIC proxy files are properly converted to RST given the recent updates (#1484, #1245).

**Comment by sarakonradi** [comment]
@jougs I discussed this PR with @terhorstd in our recent doc meeting. We thought that, other than fixing the merge conflicts, it might make more sense to bring #1374 to an end soon and to address the improvement of the MUSIC proxy and rate-based models in a follow-up PR straight afterwards. What do you think? 

Otherwise, I am happy sending another PR to @jessica-mitchell.

**Comment by heplesser** [comment]
@sarakonradi I think disentangling PRs is a good idea. Release early, release often, as they say in the agile world ;).

**Comment by heplesser** [comment]
Closed, replaced by #1530.

**Pull request closed without merging**